app/javascript/components/DollarAmountRoundingComponent.js

```diff
@@ -0,0 +1,17 @@
+export default function DollarAmountRoundingComponent() {
+    const dollarInputs = document.querySelectorAll(".round-dollar-amount");
+
+    dollarInputs.forEach((inputElement) => {
+        function render() {
+            const value = inputElement.value;
+
+            if (!isNaN(value) && !isNaN(parseFloat(value))) {
+                const rounded_value = Math.round(value);
+                inputElement.value = rounded_value;
+            }
+        }
+
+        inputElement.addEventListener("focusout", render);
+        render();
```

Comment:
why do you need to invoke render here? is there some concern that clients could input numbers before the handler was attached?

Reply:
also copy-pasta, i could have sworn i tried to remove it and it stopped working but that's probably...not what happened